R/brreg-utils.R

```diff
@@ -0,0 +1,87 @@
+#' null2na
+#' @noRd
+null2na <- function(x) if (is.null(x) || length(x) == 0) NA else x
+
+#' request_brreg
+#' @noRd
+request_brreg <- function(type = c('enheter', 'kommuner'), entity = NULL) {
+  type <- match.arg(type)
+  req <- request(brreg_url)
+  if (type == 'enheter') {
+    is_valid_number <- grepl('\\d{9}', entity)
+    #
+    if (nchar(entity) != 9 && !grepl('\\D', entity) ) { #grepl('\\d', entity) && grepl('\\D', entity)
+      msg = '{entity} is not a 9-digit organization number'
+      cli::cli_abort(msg)
+    }
+    if (is_valid_number) {
+      req <- req |>
+        req_url_path_append('enheter') |>
+        req_url_path_append(entity)
+    } else {
+      req <- req |>
+        req_url_path_append('enheter') |>
+        req_url_query('navn' = entity)
+    }
+  } else if (type == 'kommuner') {
+    req <- req |>
+      req_url_path_append('kommuner') |>
+      req_url_query(size = 450)
+  }
+  send_query(req)
+}
+
+#' parse_brreg_entity
+#' @noRd
+parse_brreg_entity <- function(parsed) {
+  if ('organisasjonsnummer' %in% names(parsed)) {
+    df <- parse_brreg_entity_(parsed)
+  } else if ('_embedded' %in% names(parsed)) {
+    # NOTE: Old code from riksrevR. Not sure this is needed.
```

Comment:
Not sure if this is needed either. The API should return a 404 if an entity is not found, not an empty result.

Reply:
Good point. I'll take an extra look at what happens, and when this is executed. Although there is a subtle difference since we are handling both names and organization numbers in the same function, and these responses don't allways act in the same way.